### PR TITLE
JOINDIN-649 Add in CRUD for talk links

### PR DESCRIPTION
**Pull request by exussum12**
Delete, Post, Put, Get added
Changed the structure of the test to make smaller tests

Had to adjust the autoloader because of this

All lines changes are covered by tests

### CODE REVIEW

**Comment by railto** [comment]
Ok to test

**Comment by exussum12** [comment]
Sorry its a little bigger than I thought. This PR contains both

File: db/patch69.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  FOREIGN KEY (`talk_type`) REFERENCES `joindin`.`talk_link_types`(`ID`)
+);
+
+
+INSERT INTO `talk_link_types`(`ID`,`type`,`display_name`) VALUES
+(1,'slides_link'),
+(2,'video_link'),
+(3,'audio_link'),
+(4,'code_link'), #github / bitbucket links
+(5,'joindin_link'); #Links to another talk on the same topic by the speaker
```
Comment (exussum12) [inline db/patch69.sql]: May end up changing this to give more info in the links back

**Comment by railto** [comment]
ok to test

**Comment by railto** [comment]
@exussum12 can you have a look at the conflicts when you get a chance please

**Comment by railto** [comment]
Note, the db patch number will need to be increased to 70

**Comment by liam-wiltshire** [comment]
@exussum12 - just to confirm, as far as I can tell, I can close #390 as the commits are contained within this? Have you got a little time to go through and sort the conflicts? Would be great to get this merged :)

**Comment by exussum12** [comment]
Thats right, all is in here, Sorry was going to look after I added to the front end. Will fix them now!

**Comment by exussum12** [comment]
Was a bit of a messy merge. Can someone please double check I didnt break anything

File: db/patch70.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  `talk_type` INT NOT NULL,
+  `url` TEXT NOT NULL,
+  `timestamp` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  PRIMARY KEY (`ID`),
+  FOREIGN KEY (`talk_id`) REFERENCES `joindin`.`talks`(`ID`),
+  FOREIGN KEY (`talk_type`) REFERENCES `joindin`.`talk_link_types`(`ID`)
+);
+
+
+INSERT INTO `talk_link_types`(`ID`,`type`,`display_name`) VALUES
```
Comment (liam-wiltshire) [inline db/patch70.sql]: @exussum12 - looks like we a missing the `display_name` values below - patch won't apply :)

Comment (exussum12) [inline db/patch70.sql]: Great spot, Fixed :)

**Comment by exussum12** [comment]
Sorry found a small issue with this, I have committed the fix on 651

**Pull request merged**